aten/src/ATen/core/jit_type.h

```diff
@@ -552,22 +567,23 @@ struct CAFFE2_API TensorType : public Type {
 
   TensorTypePtr dimensionedOnly() const {
     auto copy = clone();
-    copy->sizes_ = VaryingShape(sizes().size());
-    copy->strides_ = VaryingShape(sizes().size());
+    copy->sizes_ = VaryingShape<ShapeSymbol>(sizes().size());
+    copy->strides_ = VaryingShape<int64_t>(sizes().size());
+    copy->stride_indices_ = VaryingShape<size_t>(sizes().size());
+    copy->contiguity_ = VaryingShape<bool>(sizes().size());
     return copy;
   }
 
   TensorTypePtr contiguous() const {
     auto cloned = clone();
-    if (auto concrete_sizes = sizes().concrete_sizes()) {
-      cloned->strides_ = VaryingShape(contiguousStridesOf(*concrete_sizes));
-    } else  {
-      cloned->strides_ = VaryingShape(sizes().size());
-    }
+    TORCH_INTERNAL_ASSERT(sizes().concrete_sizes().has_value());
+    cloned->strides_ = VaryingShape<int64_t>(contiguousStridesOf(*sizes().concrete_sizes()));
     return cloned;
   }
 
-  TensorTypePtr merge(TensorTypePtr other) const;
+  const VaryingShape<ShapeSymbol>& symbolic_sizes() const;
+
+  TensorTypePtr merge(TensorTypePtr other, bool merge_sizes = true) const;
```

Comment:
`merge_sizes` is unused, its a bit confusing to have it. If you want to not merge sizes, you can just blank out the sizes after or before the merge.

Reply:
it's used in the next PR since to avoid running the default logic for `VaryingShape<ShapeSymbol>` since the merging logic will be in `profiling_record.cpp`

Reply:
That doesn't sound like a good separation of concerns, but it is difficult to suggest something better because its use isn't in this PR. Maybe remove it from here and put in the PR that actually uses it. I am skeptical that this is the best way to represent something, because this boolean flag essentially makes `merge` not perform a merge, which is an abstraction leak issue.